scripts/checks.sh

```diff
@@ -439,6 +439,46 @@ function check_service_restarts()
 	fi
 }
 
+function check_image_corruption()
+{
+	local images output result="${GOOD}"
+	local corrupted=()
+	local timeout=()
+	local -i corrupted_count timeout_count
+	# TODO: this command is filtering out any images with a size <1Kb (for delta-based images)
+	images=$(${TIMEOUT_CMD} ${ENG} image ls --format "{{.ID}} {{.Size}}" | awk '/[0-9]+B/{next;}{print $1}' | sort -u)
+
+	if (( ${#images[@]} > 0 )); then
+		for i in ${images};
+		do
+			# TODO: the timeout here is probably too short to be effective in most cases, but let's gather
+			# data and change if necessary
+			if ! ${TIMEOUT_CMD} ${ENG} save "${i}" > /dev/null ; then
+				# this retval indicates a timeout
+				if (( $? = 124 )); then
+					timeout_count+=1
+					timeout+=("${i}")
+				else
+					corrupted_count+=1
+					corrupted+=("${i}")
+				fi
+			fi
+		done
+		if (( "${corrupted_count}" != 0 )); then
+			output+="Some images may be corrupted: ${corrupted[*]}\n"
```

Comment:
Nit: IIUC, in Bash arrays these two lines are equivalent:

```
${array[@]}  # All items in array
${array[*]} # Also all items in array
```

The first syntax is used at line 451; is it worth sticking with that syntax for consistency, here and at line 472?

Reply:
There are some subtle differences between the two iterators with respect to splitting and quoting (same as `$@` vs `$*`, see https://tldp.org/LDP/abs/html/internalvariables.html#APPREF). Shellcheck doesn't always like this either: https://github.com/koalaman/shellcheck/wiki/SC2124.

Reply:
TIL. Thanks for the links!